File: 5-Problem-A-Day/14. Longest Common Prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        if (strs.length == 0)
            return "";
        StringBuilder ans = new StringBuilder();
        boolean diff = false;
        int index = 0;
        while (!diff && index < strs[0].length()) {
            char c = strs[0].charAt(index);
            for (int i = 1; i < strs.length; i++) {
                if (index >= strs[i].length() || strs[i].charAt(index) != c) {
                    diff = true;
                    break;
                }
            }
            if (!diff)
                ans.append(c);
            index += 1;
        }
        return ans.toString();
    }
}

File: 2024/14. Longest Common Prefix.cpp
class Solution {
public:
    string longestCommonPrefix(vector<string>& strs) {
        string prefix;
        int index = 0;
        while (true) {
            for (int i = 0; i < strs.size(); i++) {
                if (index >= strs[i].length() || strs[i][index] != strs[0][index]) {
                    return prefix;
                }
            }
            prefix += strs[0][index];
            index += 1;
        }
        return prefix;
    }
};